src/transformers/models/deberta_v2/tokenization_deberta_v2.py

```diff
@@ -336,7 +341,27 @@ def convert_ids_to_tokens(self, ids):
 
     def decode(self, tokens, start=-1, end=-1, raw_text=None):
         if raw_text is None:
-            return self.spm.decode_pieces([t for t in tokens])
+            current_sub_tokens = []
+            out_string = ""
+            prev_is_special = False
+            seen_unk = False
+            for token in tokens:
+                # make sure that special tokens are not decoded using sentencepiece model
+                if token in self.special_tokens:
+                    if not prev_is_special and not seen_unk:
+                        out_string += " "
+                    if seen_unk:
+                        seen_unk = False
+                    out_string += self.spm.decode_pieces(current_sub_tokens) + token
+                    prev_is_special = True
+                    current_sub_tokens = []
+                    if token == self.special_tokens[2]:  # if token == <unk>
+                        seen_unk = True
```

Comment:
I'm going to sound annoying but I think it's risky to do it this way because there's nothing that requires the unknown token to be in the third position. Do you remember if another tokenizer is in the same situation?

Reply:
No other tokenizer had this issue during my work on this PR. I agree it's risky, and it's not even aligned with the comment that appears in the `__init__` method. That's the problem with being an "outsider", I don't know if there's a mistake in the fast tokenizer or am I missing something 😅 The behavior of `<unk>` is different in DeBERTa v2 than in the other SP tokenizers.

Reply:
This special code should indeed be removed. I think it's okay to exclude Deberta-v2 from the new test, decode as it makes the most sense even if it breaks the compatibility with the fast tokenizer if we think there is a bug there. We don't guarantee 100% match between the two in all situations.